apps/shipments/admin/shipment.py

```diff
@@ -151,6 +151,16 @@ def route_link(self, obj):
 class HistoricalShipmentAdmin(BaseModelHistory, ShipmentAdmin):
     readonly_fields = [field.name for field in Shipment._meta.get_fields()]
 
+    def get_fieldsets(self, request, obj=None):
```

Comment:
Your PR description says you are removing 'route_link' from fieldsets, yet it's not referenced anywhere in this class. It looks like you've made some assumptions about the fieldsets ordering here and hardcoded some array indexing.